contracts/parent-chain/contract-upgrades/EspressoArbOSUpgrade.sol

```diff
@@ -0,0 +1,34 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import "forge-std/Script.sol";
+
+import "@arbitrum/nitro-contracts/src/precompiles/ArbOwner.sol";
+
+/// @notice should be included in an operation batch in the L1 timelock along with SetWasmModuleRootAction
+contract UpgradeArbOSVersionAtTimestampAction {
+    uint64 public immutable newArbOSVersion;
+    uint64 public immutable upgradeTimestamp;
+
+    constructor(uint64 _newArbOSVersion, uint64 _upgradeTimestamp) {
```

Comment:
do we need uint64 for the arbOSVersion?

Reply:
I would argue that most of the way this is done doesn't "need" to happen this way, but I wrote these contracts to broadly conform with the guidance in the arbitrum governance repo [here](https://github.com/ArbitrumFoundation/governance/blob/main/src/gov-action-contracts/README.md)

For the sake of time we aren't using address registries and are instead using env variables for the generic and proposal specific version of the contracts. I'm just doing it this way to minimize the amount of future work to convert this to a reasonable proposal for the Arbitrum DAO

Reply:
ok great!